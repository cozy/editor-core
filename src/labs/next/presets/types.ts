import { FeatureFlags } from '../../../plugins/feature-flags-context/types';

export interface EditorPresetProps {
  excludes?: Set<string>;
  experimental?: Array<string>;
  featureFlags?: FeatureFlags;
}
