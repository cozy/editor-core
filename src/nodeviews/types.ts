import type { Node as PMNode } from '@atlaskit/editor-prosemirror/model';

export interface ReactNodeProps {
  selected: boolean;
}

export type ProsemirrorGetPosHandler = () => number | undefined;
export type getPosHandler = getPosHandlerNode | boolean;
export type getPosHandlerNode = () => number | undefined;
export type ReactComponentProps = { [key: string]: unknown };
export type ForwardRef = (node: HTMLElement | null) => void;
export type shouldUpdate = (nextNode: PMNode) => boolean;
