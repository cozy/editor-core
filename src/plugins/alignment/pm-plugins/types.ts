export type AlignmentState = 'start' | 'end' | 'center';
export type AlignmentPluginState = {
  align: AlignmentState;
  isEnabled?: boolean;
};
