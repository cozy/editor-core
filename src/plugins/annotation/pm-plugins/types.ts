import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';
import type {
  EditorState,
  SelectionBookmark,
} from '@atlaskit/editor-prosemirror/state';
import type { PortalProviderAPI } from '@atlaskit/editor-common/portal-provider';
import type { EditorAnalyticsAPI } from '@atlaskit/editor-common/analytics';
import type {
  Dispatch,
  EventDispatcher,
} from '@atlaskit/editor-common/event-dispatcher';
import type { InlineCommentAnnotationProvider, AnnotationInfo } from '../types';

export enum ACTIONS {
  UPDATE_INLINE_COMMENT_STATE,
  SET_INLINE_COMMENT_DRAFT_STATE,
  INLINE_COMMENT_UPDATE_MOUSE_STATE,
  INLINE_COMMENT_CLEAR_DIRTY_MARK,
  ADD_INLINE_COMMENT,
  INLINE_COMMENT_SET_VISIBLE,
  CLOSE_COMPONENT,
  SET_SELECTED_ANNOTATION,
}

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  provider: InlineCommentAnnotationProvider;
  editorAnalyticsAPI: EditorAnalyticsAPI | undefined;
}
export interface InlineCommentMouseData {
  isSelecting: boolean;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.UPDATE_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: InlineCommentMouseData;
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_CLEAR_DIRTY_MARK }
  | { type: ACTIONS.CLOSE_COMPONENT }
  | {
      type: ACTIONS.ADD_INLINE_COMMENT;
      data: {
        drafting: boolean;
        inlineComments: InlineCommentMap;
        editorState: EditorState;
        selectedAnnotations: AnnotationInfo[];
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_SET_VISIBLE; data: { isVisible: boolean } }
  | {
      type: ACTIONS.SET_SELECTED_ANNOTATION;
      data: { selectedAnnotations: AnnotationInfo[] };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  selectedAnnotations: AnnotationInfo[];
  dirtyAnnotations?: boolean;
  mouseData: InlineCommentMouseData;
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark;

  // Denotes if annotations are allowed to be create on empty nodes or nodes of whitespace (Confluence spec)
  disallowOnWhitespace: boolean;

  // Allow users to hide inline comments during editing
  isVisible: boolean;
  skipSelectionHandling: boolean;
};
