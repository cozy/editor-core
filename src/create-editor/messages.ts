import { defineMessages } from 'react-intl-next';
export const editorMessages = defineMessages({
  editorAssistiveLabel: {
    id: 'fabric.editor.editorAssistiveLabel',
    defaultMessage: 'Main content area, start typing to enter text.',
    description:
      'Text that is read out by screen reader when the main editor is in focus',
  },
});
