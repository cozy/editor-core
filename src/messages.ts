// Common Translations will live here
import commonMessages, {
  linkMessages,
  linkToolbarMessages,
  statusMessages,
  dateMessages,
} from '@atlaskit/editor-common/messages';

export { linkMessages };
export { statusMessages };
export { dateMessages };
export { linkToolbarMessages };

export default commonMessages;
