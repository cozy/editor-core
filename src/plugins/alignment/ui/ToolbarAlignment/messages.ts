import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  alignment: {
    id: 'fabric.editor.alignment',
    defaultMessage: 'Alignment',
    description: 'Aligns text',
  },
});
