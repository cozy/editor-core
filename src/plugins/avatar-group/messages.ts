import { defineMessages } from 'react-intl-next';

export const avatarGroupMessages = defineMessages({
  editors: {
    id: 'fabric.editor.editors',
    defaultMessage: 'Editors',
    description:
      'classifying the people that are currently editing the document',
  },
});
