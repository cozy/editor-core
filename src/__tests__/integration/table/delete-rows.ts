import { IntlProvider } from 'react-intl';
import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import { editable, getDocFromElement, fullpage } from '../_helpers';
import { documentWithMergedCells } from './__fixtures__/merged-rows-and-cols-document';
import { nestedInExtension } from './__fixtures__/nested-in-extension';
import { TableCssClassName as ClassName } from '../../../plugins/table/types';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import messages from '../../../plugins/table/ui/messages';
import { deleteRow } from '../../__helpers/page-objects/_table';

BrowserTestCase(
  'Should delete merged rows from contextual menu and append missing cells to the table',
  { skip: ['edge', 'firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    const intlProvider = new IntlProvider({ locale: 'en' });
    const { intl } = intlProvider.getChildContext();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(documentWithMergedCells),
      allowTables: {
        advanced: true,
      },
    });

    const controlSelector = `tbody tr:nth-child(2) td:first-child`;
    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);

    const contextMenuTriggerSelector = `.${ClassName.CONTEXTUAL_MENU_BUTTON}`;
    await page.waitForSelector(contextMenuTriggerSelector);
    await page.click(contextMenuTriggerSelector);

    const message = await intl.formatMessage(messages.removeRows, {
      0: 1,
    });
    const contextMenuItemSelector = `span=${message}`;
    await page.waitForSelector(contextMenuItemSelector);
    await page.click(contextMenuItemSelector);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Should delete merged rows from contextual menu and decrement rowspan of the spanning cell',
  { skip: ['edge', 'firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    const intlProvider = new IntlProvider({ locale: 'en' });
    const { intl } = intlProvider.getChildContext();

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(documentWithMergedCells),
      allowTables: {
        advanced: true,
      },
    });

    const controlSelector = `tbody tr:nth-child(3) td:first-child`;
    await page.waitForSelector(controlSelector);
    await page.click(controlSelector);

    const contextMenuTriggerSelector = `.${ClassName.CONTEXTUAL_MENU_BUTTON}`;
    await page.waitForSelector(contextMenuTriggerSelector);
    await page.click(contextMenuTriggerSelector);

    const message = await intl.formatMessage(messages.removeRows, {
      0: 1,
    });
    const contextMenuItemSelector = `span=${message}`;
    await page.waitForSelector(contextMenuItemSelector);
    await page.click(contextMenuItemSelector);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Should delete a row when table is nested inside bodied extension',
  { skip: ['edge', 'firefox', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(nestedInExtension),
      allowTables: {
        advanced: true,
      },
      allowExtension: true,
    });

    await deleteRow(page, 1);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
