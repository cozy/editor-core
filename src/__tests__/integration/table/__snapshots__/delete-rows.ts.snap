// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should delete a row when table is nested inside bodied extension 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "extensionKey": "bodied-eh",
        "extensionType": "com.atlassian.confluence.macro.core",
        "layout": "default",
        "parameters": Object {
          "macroMetadata": Object {
            "placeholder": Array [
              Object {
                "data": Object {
                  "url": "",
                },
                "type": "icon",
              },
            ],
          },
          "macroParams": Object {},
        },
        "text": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__autoSize": false,
            "isNumberColumnEnabled": false,
            "layout": "default",
          },
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      242,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      308,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      176,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
              ],
              "type": "tableRow",
            },
            Object {
              "content": Array [
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      242,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      308,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
                Object {
                  "attrs": Object {
                    "background": null,
                    "colspan": 1,
                    "colwidth": Array [
                      176,
                    ],
                    "rowspan": 1,
                  },
                  "content": Array [
                    Object {
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
              ],
              "type": "tableRow",
            },
          ],
          "type": "table",
        },
      ],
      "type": "bodiedExtension",
    },
  ],
  "type": "doc",
}
`;

exports[`Should delete merged rows from contextual menu and append missing cells to the table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "1",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#deebff",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#eae6ff",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#ffebe6",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#ffebe6",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#e3fcef",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#e3fcef",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#deebff",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#eae6ff",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`Should delete merged rows from contextual menu and decrement rowspan of the spanning cell 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "1",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#deebff",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#eae6ff",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#fffae6",
                "colspan": 3,
                "colwidth": Array [
                  98,
                  275,
                  83,
                ],
                "rowspan": 2,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#e3fcef",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#e3fcef",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  151,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#deebff",
                "colspan": 1,
                "colwidth": Array [
                  98,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "2",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#e6fcff",
                "colspan": 1,
                "colwidth": Array [
                  275,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "3",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#eae6ff",
                "colspan": 1,
                "colwidth": Array [
                  83,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "4",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": "#ffffff",
                "colspan": 1,
                "colwidth": Array [
                  152,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "5",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
